### PR TITLE
FP comparisons + conversions

### PR DESCRIPTION
**Pull request by mp-17**
This is the merge of the pull requests https://github.com/pulp-platform/ara/pull/19 and https://github.com/pulp-platform/ara/pull/14

### Added

- Support for FP conversions
- Support for FP comparisons

### Changed

- Latency of the 16-bit multiplier, from 0 to 1
- Introduced a spill-register at the edge of the lanes, to cut the `mask_ready_o` timing-critical path.

## Checklist

- [x] Automated tests pass
- [x] Changelog updated
- [ ] Code style guideline is observed

Please check our [contributing guidelines](CONTRIBUTING.md) before opening a Pull Request.

### CODE REVIEW

**Comment by mp-17** [comment]
Rebased after merging https://github.com/pulp-platform/ara/pull/40

File: hardware/include/ara_pkg.sv
Comment (suehtamacv) [inline hardware/include/ara_pkg.sv:61]: I assume the critical path goes through this multiplier when synthesizing this Ara? Maybe another incremental compile run would allow us to remove it...

File: hardware/src/lane/lane.sv
Comment (suehtamacv) [inline hardware/src/lane/lane.sv:99]: Vertical alignment. Please also use a more descriptive name for the instance. Maybe `i_mask_spill_register`?

File: hardware/src/lane/lane_sequencer.sv
Comment (suehtamacv) [inline hardware/src/lane/lane_sequencer.sv:469]: No need to use a ternary operator.

File: hardware/src/lane/lane_sequencer.sv
Comment (suehtamacv) [inline hardware/src/lane/lane_sequencer.sv:497]: Which instruction needs both the `Alu` and the `MulFPU` queues?

File: hardware/src/lane/vector_fus_stage.sv
Comment (suehtamacv) [inline hardware/src/lane/vector_fus_stage.sv:87]: Vertical alignment.

File: hardware/src/lane/vector_fus_stage.sv
Comment (suehtamacv) [inline hardware/src/lane/vector_fus_stage.sv:122]: Vertical alignment.

File: hardware/src/lane/vector_fus_stage.sv
Comment (suehtamacv) [inline hardware/src/lane/vector_fus_stage.sv:162]: Vertical alignment.

File: hardware/src/lane/vmfpu.sv
Comment (suehtamacv) [inline hardware/src/lane/vmfpu.sv:175]: Could we rearrange the operations so that they are all contiguous?

File: hardware/src/lane/vmfpu.sv
Comment (suehtamacv) [inline hardware/src/lane/vmfpu.sv:1015]: Please wrap the lines to 100 characters.

**Review by suehtamacv — CHANGES_REQUESTED** [review]
Looks good, thanks Perotti. I have a few nitpicks, and I would also like to merge this PR after we merge #50, since that PR implements an improved and faster CI.

File: hardware/src/lane/lane_sequencer.sv
Comment (mp-17) [reply hardware/src/lane/lane_sequencer.sv:502]: No one, indeed we push only in the correct opqueue: 

`if (pe_req_i.op inside {[VMFEQ:VMFGE]})` for the ALU

`if !(pe_req_i.op inside {[VMFEQ:VMFGE]})` for the VMFPU

File: hardware/include/ara_pkg.sv
Comment (mp-17) [reply hardware/include/ara_pkg.sv:61]: Agree. Can we postpone this to the optimization phase? When the HW is more mature we can try to optimize the latency of the units. Or do you prefer to do it now?

**Review by suehtamacv — APPROVED** [review]
LGTM! Waiting for the CI.

**Pull request merged**
